satpy/tests/reader_tests/test_grib.py

```diff
@@ -207,6 +207,30 @@ def test_file_pattern(self):
         files = r.select_files_from_pathnames(filenames)
         self.assertEqual(len(files), 4)
 
+    def _round_trip_projection_lonlat_check(self, area):
+        """Check that X/Y coordinates can be transformed multiple times.
+
+        Many GRIB files include non-standard projects that work for the
+        initial transformation of X/Y coordinates to longitude/latitude,
+        but may fail in the reverse transformation. For example, an eqc
+        projection that goes from 0 longitude to 360 longitude. The X/Y
+        coordinates may accurately go from the original X/Y metered space
+        to the correct longitude/latitude, but transforming those coordinates
+        back to X/Y space will produce the wrong result.
+
+        """
+        if not hasattr(area, 'crs'):
+            # skip this check if we aren't in pyproj 2.0+
+            return
+
+        from pyproj import Proj
+        p = Proj(area.crs)
+        x, y = area.get_proj_vectors()
+        lon, lat = p(x, y, inverse=True)
+        x2, y2 = p(lon, lat)
+        np.testing.assert_almost_equal(x, x2)
+        np.testing.assert_almost_equal(y, y2)
+
```

Comment:
I think this deserves its own test.

Reply:
I guess, but there will be more projections in the user (I'm sure) that will need this exact check. I suppose I could duplicate parts of the existing tests (with properly refactored helper methods of course) that would allow this to be specifically tested.

Reply:
Now you've done it. I pytest parameterize'd them.